### PR TITLE
Fixes #35006 - Remove runcible and conclude Pulp 2 removal

### PR DESCRIPTION
**Pull request by ianballou**
#### What are the changes introduced in this pull request?
Rips out all of the remaining Pulp 2 related code.

Some remains, like code relating to Docker push. We need to reimplement it with Pulp 3, so I figured that having the old code would serve as a good starting point. The APIs for users to interact with the code have been commented out, however.

#### Considerations taken when implementing this change?
A lot of things regarding sync status were removed, and it appears that none of them were in use. The sync status page still works. However, I'd like some help double checking that that all still works. Sync plans for example, would be good to test.

Similarly, the container registry had some changes. Looks like all is set, but some more eyes on it would be good. Perhaps I could further simplify some of it as well.

#### What are the testing steps for this pull request?
1) Make sure syncing all still works. This includes sync plans.
2) Make sure that the container registry still works.
3) Some package group related code was removed. Double check that there aren't floating APIs mentioning package group deletion or creation.

### CODE REVIEW

**Comment by ianballou** [comment]
[test katello]

**Comment by ianballou** [comment]
[test katello]

File: lib/katello/repository_types/README.md
Comment (sjha4) [inline lib/katello/repository_types/README.md:1]: Should we update this file for pulp3 content type registration?

**Review by sjha4 — APPROVED** [review]
One minor comment around repository type registration documentation. Other than that worked as expected in my testing. Ship it! 👍🏼

**Comment by sjha4** [comment]
Also can we add a redmine issue for the last commit and also don't forget to squash and merge 😸

**Comment by ekohl** [comment]
I'd actually NOT squash. The commits were rather carefully written and now it's at least feasible to see differences. If you squash it, git blame becomes much harder.

File: lib/katello/repository_types/README.md
Comment (ianballou) [reply lib/katello/repository_types/README.md:1]: yeah not a bad idea. I didn't realize it got deleted.

**Comment by ianballou** [comment]
@sjha4 I edited the commit messages and re-wrote the repo type file README.

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:63]: lol, cute

It would be nice for the "friends" to be fully documented too in the future, but not necessary to block on it.... maybe throw a card on the backlog to revisit?

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:73]:
```suggestion
There are a number of other classes that need to be defined, so follow the "Full example" above to ensure that the new type has
```

Or even better, put that under a heading like `# Full Example` so we can link directly to it here

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:74]: today I learned: https://stackoverflow.com/questions/41513565/markdown-and-line-wrapping

:+1:

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:90]:
```suggestion
The `default_managed_content_type` method sets the Rpm content type as the default content type, which
the repository type will act upon when another content type is not explicitly declared. For example, a yum upload
will assume that the uploaded content is of Rpm type unless otherwise specified.
```

Trying to maintain the precision of what you have while improving the flow a bit, which is not easy. Feel free to disregard

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:92]:
```suggestion
The `content_type` method defines a content type to be used by the repository.  The first argument is the Active Record model class representing the content type.  This should extend the `Katello::Concerns::PulpDatabaseUnit` class.
```

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:95]:
```suggestion
* pulp3_service_class - The service class representing this content type
```

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:100]: Should `index_on_pulp3` be removed as part of this PR, then? As of this review it is still used in `lib/katello/repository_types/yum.rb` and maybe some others (I didn't check)

Then we could also remove the reference to it here

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:99]:
```suggestion
* index - Content units of this type will be indexed
```

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:117]:
```suggestion
This option declares that this content type supports the CheckMatchingContent middleware, which removes the need to republish repositories in pulp when content was not changed.
```

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:124]:
```suggestion
The `generic_content_type` and `generic_remote_option` methods exist to define all other attributes that are needed to
```

File: lib/katello/repository_types/README.md
Comment (wbclark) [inline lib/katello/repository_types/README.md:119]: Huge :+1: for adding this section

**Review by wbclark — COMMENTED** [review]
In-line feedback is mostly minor so take a look and use what you like.

The bigger items would be:

1. I think the rubocop workaround got added to the wrong commit
2. Unless this was already discussed and agreed upon, shouldn't the deprecated `index_on_pulp3` be outright removed?
3. Didn't comment on this one in-line, but there's mixed usage of `:key => value` and `key: value` throughout the README but it might be nice to just use the newer syntax always

File: lib/katello/repository_types/README.md
Comment (wbclark) [reply lib/katello/repository_types/README.md:117]: (Though I do love me some archaic "whereby" usage)

File: lib/katello/repository_types/README.md
Comment (ianballou) [reply lib/katello/repository_types/README.md:14]: That's Pulp 3 generated bindings for you :)

File: lib/katello/repository_types/README.md
Comment (ianballou) [reply lib/katello/repository_types/README.md:63]: Yeah the generic content type framework would need a full-blown technical docs effort.

File: lib/katello/repository_types/README.md
Comment (ianballou) [reply lib/katello/repository_types/README.md:90]: Yeah most of this is copy/pasted from the old readme that was accidentally deleted in an older commit.

**Review by chris1984 — APPROVED** [review]
Looks fine, @wbclark want to give it a final review so we can get this in today before branching?

**Review by wbclark — APPROVED** [review]
I think one of the commits needs to say `Fixes #35006` instead of `Refs #35006` for the sake of Redmine automation.

That has always seemed to work in the past when using 'Rebase and Merge'

Looks great BTW, thanks @ekohl and @ianballou !

**Comment by ekohl** [comment]
@wbclark not really, since this doesn't remove ALL Pulp 2 code. This is just part 1.

**Comment by wbclark** [comment]
> @wbclark not really, since this doesn't remove ALL Pulp 2 code. This is just part 1.
> 
> Edit: though I now see it's far enough to remove runcible from the gemspec, so that's a big goal. Perhaps this is complete then.

I think @ianballou can give us the definitive answer :)

In case the intent is to keep all commits as `Refs` to hold the Redmine open for follow-up work after this is merged, just change the PR title itself to `Refs` as well then, to keep it consistent.

**Comment by ekohl** [comment]
We can make it "Remove runcible dependency", which it certainly fixes.

**Comment by ianballou** [comment]
Ah in all my rebasing I forgot that "Fixes". I'll fix the Fixes.

**Comment by ianballou** [comment]
This is supposed to be the final Pulp 2 removal.

**Pull request merged**
